Core/include/Acts/Utilities/Intersection.hpp

```diff
@@ -39,145 +39,227 @@ inline std::ostream& operator<<(std::ostream& os, IntersectionStatus status) {
+  /// @param object is the object to be instersected
   template <typename T = representation_t,
             std::enable_if_t<std::is_same<T, object_t>::value, int> = 0>
-  ObjectIntersection(const Intersection3D& sInter, const object_t* sObject)
-      : intersection(sInter), object(sObject), representation(sObject) {}
+  constexpr ObjectMultiIntersection(const MultiIntersection3D& intersections,
+                                    const object_t* object)
+      : m_intersections(intersections),
+        m_object(object),
+        m_representation(object) {}
 
   /// Object intersection
   ///
-  /// @param sInter is the intersection
-  /// @param sObject is the object to be instersected
-  /// @param sRepresentation is the object representation
-  ObjectIntersection(const Intersection3D& sInter, const object_t* sObject,
-                     const representation_t* sRepresentation)
-      : intersection(sInter),
-        object(sObject),
-        representation(sRepresentation) {}
-
-  /// Bool() operator for validity checking
-  explicit operator bool() const { return bool(intersection); }
-
-  /// @brief Smaller operator for ordering & sorting
-  ///
-  /// This operator will ignore the alternative, but simply
-  /// order the representing intersection
-  ///
-  /// @param oi is the source intersection for comparison
-  bool operator<(
-      const ObjectIntersection<object_t, representation_t>& oi) const {
-    return (intersection < oi.intersection);
+  /// @param intersections are the intersections
+  /// @param object is the object to be instersected
+  /// @param representation is the object representation
+  constexpr ObjectMultiIntersection(const MultiIntersection3D& intersections,
+                                    const object_t* object,
+                                    const representation_t* representation)
+      : m_intersections(intersections),
+        m_object(object),
+        m_representation(representation) {}
+
+  constexpr ObjectIntersection<object_t, representation_t> operator[](
+      std::uint8_t index) const {
+    return {m_intersections[index], m_object, m_representation, index};
   }
 
-  /// @brief Greater operator for ordering & sorting
-  ///
-  /// This operator will ignore the alternative, but simply
-  /// order the representing intersection
-  ///
-  /// @param oi is the source intersection for comparison
-  bool operator>(
-      const ObjectIntersection<object_t, representation_t>& oi) const {
-    return (intersection > oi.intersection);
+  constexpr std::size_t size() const { return m_intersections.size(); }
+
+  constexpr const object_t* object() const { return m_object; }
+
+  constexpr const representation_t* representation() const {
+    return m_representation;
   }
 
-  /// Allow swapping the intersection and the alternative
-  void swapSolutions() { std::swap(intersection, alternative); }
-};
+  constexpr std::array<ObjectIntersection<object_t, representation_t>, 2>
+  split() const {
```

Comment:
Doing this, for example, with a `static_vector`, could explicitly communicate if we have one or two intersections, and not require checking if the intersection object itself is valid.